app/controllers/investigations/corrective_actions_controller.rb

```diff
@@ -61,7 +61,7 @@ def update
 
       return render :edit if @corrective_action_form.invalid?(:edit_corrective_action)
 
-      result = UpdateCorrectiveAction.call(
+      UpdateCorrectiveAction.call(
```

Comment:
```suggestion
      UpdateCorrectiveAction.call!(
```
Since we removed the `success` conditional. Throws exception on failure, otherwise we will not know if it failed and may erroneously report to the user that it succeeded.